libs/parallelism/algorithms/include/hpx/parallel/util/loop.hpp

```diff
@@ -1038,3 +1035,7 @@ namespace hpx { namespace parallel { namespace util {
         return val;
     }
 }}}    // namespace hpx::parallel::util
+
+#if defined(HPX_HAVE_DATAPAR)
+#include <hpx/parallel/datapar/loop.hpp>
+#endif
```

Comment:
What is the rationale of moving this `#include`?

Reply:
@hkaiser The CPOs are not visible when we use tag_invoke if this is at top.

Reply:
I don't think the `tag_invoke` overload you provide has to be defined only after the definition of the CPO. It should work either way.

Reply:
this is for loop_n_t CPO https://github.com/STEllAR-GROUP/hpx/blob/9d281b96351f6e35f1b456c8cda547969c5188a3/libs/parallelism/algorithms/include/hpx/parallel/util/loop.hpp#L324 

if datapar header is defined at the top then https://github.com/STEllAR-GROUP/hpx/blob/9d281b96351f6e35f1b456c8cda547969c5188a3/libs/parallelism/algorithms/include/hpx/parallel/datapar/loop.hpp#L353 here in tag_invoke, i got error during compilation saying it could not find hpx::parallel::util::detail::loop_n_t

build_error : 
https://app.circleci.com/pipelines/github/STEllAR-GROUP/hpx/7432/workflows/ff16c013-5477-4dc6-aa42-0fb2cfd46025/jobs/247902